src/NServiceBus.Core/SettingsExtensions.cs

```diff
@@ -28,24 +28,24 @@ public static string EndpointName(this ReadOnlySettings settings)
         }
 
         /// <summary>
-        /// Returns the logical address of this endpoint.
+        /// Returns the transport specific address of the shared queue name of this endpoint.
         /// </summary>
-        public static LogicalAddress LogicalAddress(this ReadOnlySettings settings)
+        public static string LocalAddress(this ReadOnlySettings settings)
         {
             Guard.AgainstNull(nameof(settings), settings);
 
             if (!settings.TryGet<ReceiveComponent.Configuration>(out var receiveConfiguration))
             {
-                throw new InvalidOperationException("LogicalAddress isn't available since this endpoint is configured to run in send-only mode.");
+                throw new InvalidOperationException("LocalAddress isn't available since this endpoint is configured to run in send-only mode.");
             }
 
-            return receiveConfiguration.LogicalAddress;
+            return receiveConfiguration.LocalAddress;
         }
 
         /// <summary>
         /// Returns the shared queue name of this endpoint.
         /// </summary>
-        public static string LocalAddress(this ReadOnlySettings settings)
+        public static string EndpointQueueName(this ReadOnlySettings settings)
```

Comment:
We should consider moving this one to the feature configuration context. But that can be done in a separate PR.

Reply:
yeah that might make sense. +1 for a separate PR.